### PR TITLE
Convert gce-FEATURE-release-x jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
[regex101](regex101.com) is my good friend 

#1230

### CODE REVIEW

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -1266,10 +1266,15 @@ def testBootstrapCIYaml(self):
...
+            'kubernetes-e2e-gci-gce',
+            'kubernetes-e2e-gce[a-z-]+release-1.[345]$'
+        ]
+
+        blacklist = [
+            'kubernetes-e2e-gce-gci-'
         ]
             
-        is_modern = lambda name: any(re.match(w, name) for w in whitelist)
+        is_modern = lambda name: any(re.match(w, name) for w in whitelist) and all(re.match(b, name) is None for b in blacklist)
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Favor implicit comparisons over `is None`

https://google.github.io/styleguide/pyguide.html?showone=True/False_evaluations#True/False_evaluations

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: This might be better

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: If you have a spare minute, how do I get this regex right? 
`kubernetes-e2e-gce^(?!gci)[a-z-]+release-1.[345]$`

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: oh I don't need `^` :-p

**Pull request merged**